proxy/http2/Http2ClientSession.cc

```diff
@@ -326,6 +327,9 @@ Http2ClientSession::main_event_handler(int event, void *edata)
     break;
 
   case VC_EVENT_WRITE_COMPLETE:
+    if (this->get_half_close_flag()) {
```

Comment:
I agree that there is no guarantee, but we should try to process the outstanding streams as long as possible. Two questions.
1) Once a GOAWAY frame is sent, shouldn't we stop accepting new streams?
2) should we close the session only when there are no outstanding streams?